Switch torch.save to zipfile serialization and swap quantization to that

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#39460 Switch torch.save to zipfile serialization and swap quantization to that**

Differential Revision: [D21865748](https://our.internmc.facebook.com/intern/diff/D21865748)

Commit:
Switch torch.save to zipfile serialization and swap quantization to that

[ghstack-poisoned]

Commit:
Update on "Switch torch.save to zipfile serialization and swap quantization to that"


Differential Revision: [D21865748](https://our.internmc.facebook.com/intern/diff/D21865748)

[ghstack-poisoned]

Commit:
Update on "Switch torch.save to zipfile serialization and swap quantization to that"


Differential Revision: [D21865748](https://our.internmc.facebook.com/intern/diff/D21865748)

[ghstack-poisoned]

Commit:
Update on "Switch torch.save to zipfile serialization and swap quantization to that"


Differential Revision: [D21865748](https://our.internmc.facebook.com/intern/diff/D21865748)

[ghstack-poisoned]

Commit:
Update on "Switch torch.save to zipfile serialization and swap quantization to that"


Differential Revision: [D21865748](https://our.internmc.facebook.com/intern/diff/D21865748)

[ghstack-poisoned]

jerryzh168 commented on torch/nn/quantized/modules/conv.py:
```diff
@@ -92,11 +92,6 @@ def _save_to_state_dict(self, destination, prefix, keep_vars):
 
     @torch.jit.export
     def __getstate__(self):
```
can we remove these now? since packed params are serializable

jamesr66a replied:
I'd rather do that in a separate PR. This PR changes behavior in a ton of places and if it gets reverted it's better to have a minimal number of changes in here

jerryzh168 reviewed (approved):
This is great, thanks!

Commit:
Update on "Switch torch.save to zipfile serialization and swap quantization to that"


Differential Revision: [D21865748](https://our.internmc.facebook.com/intern/diff/D21865748)

[ghstack-poisoned]

Commit:
Update on "Switch torch.save to zipfile serialization and swap quantization to that"


Differential Revision: [D21865748](https://our.internmc.facebook.com/intern/diff/D21865748)

[ghstack-poisoned]
